src/nyc_trees/js/src/progressPage.js

```diff
@@ -14,15 +14,48 @@ var progressMap = mapModule.create({
         legend: true,
         search: true
     }),
+    tileLayer = null,
+    grid = null,
+    selectedLayer = null,
 
     dom = {
+        modeDropdown: '.dropdown',
+        modeButton: '.btn',
+        modeChoice: '.dropdown li a',
+        legend: '#legend',
         actionBar: '#action-bar'
     },
     $actionBar = $(dom.actionBar);
 
-mapModule.addTileLayer(progressMap);
+$(dom.modeChoice).click(onModeChanged);
 
-var grid = mapModule.addGridLayer(progressMap),
+$(dom.modeChoice).first().trigger('click');
+
+function onModeChanged(e) {
+    var $mode = $(e.currentTarget),
+        tileUrl = $mode.data('tile-url'),
+        gridUrl = $mode.data('grid-url'),
+        bounds = $mode.data('bounds');
+
+    // Shown chosen mode on dropdown button
+    $mode.parents(dom.modeDropdown).find(dom.modeButton).text($mode.text());
+
+    // Show appropriate legend block
+    $(dom.legend).children().hide();
+    $($mode.attr('href')).show();
```

Comment:
I just paired with @jfrankl on cleaning up the progress map legend.

It might be best to let him merge those changes before this PR gets merged, to avoid conflicts.